third_party/2and3/pynamodb/attributes.pyi

```diff
@@ -0,0 +1,137 @@
+from typing import Any, Callable, Dict, Optional, Text, Type, Union, Set
+
+from datetime import datetime
+
+class Attribute:
+    attr_name = ...  # type: Any
+    attr_type = ...  # type: Any
+    null = ...  # type: bool
+    default = ...  # type: Any
+    is_hash_key = ...  # type: Any
+    is_range_key = ...  # type: Any
+    def __init__(self, hash_key: bool = ..., range_key: bool = ..., null: Optional[bool] = ..., default: Optional[Union[Any, Callable[..., Any]]] = ..., attr_name: Optional[Text] = ...) -> None: ...
+    def __set__(self, instance, value): ...
+    def __get__(self, instance, owner): ...
+    def serialize(self, value): ...
+    def deserialize(self, value): ...
+    def get_value(self, value): ...
+
+class SetMixin:
+    def serialize(self, value): ...
+    def deserialize(self, value): ...
+
+class BinaryAttribute(Attribute):
+    attr_type = ...  # type: Any
+    def serialize(self, value): ...
+    def deserialize(self, value): ...
+    def __init__(self, hash_key: bool = ..., range_key: bool = ..., null: Optional[bool] = ..., default: Optional[Union[bytes, Callable[..., bytes]]] = ..., attr_name: Optional[Text] = ...) -> None: ...
+    def __set__(self, instance, value: bytes): ...
+    def __get__(self, instance, owner) -> bytes: ...
+
+class BinarySetAttribute(SetMixin, Attribute):
+    attr_type = ...  # type: Any
+    null = ...  # type: bool
+    def serialize(self, value): ...
+    def deserialize(self, value): ...
+    def __init__(self, hash_key: bool = ..., range_key: bool = ..., null: Optional[bool] = ..., default: Optional[Union[Set[bytes], Callable[..., Set[bytes]]]] = ..., attr_name: Optional[Text] = ...) -> None: ...
+    def __set__(self, instance, value: Set[bytes]): ...
+    def __get__(self, instance, owner) -> Set[bytes]: ...
+
+class UnicodeSetAttribute(SetMixin, Attribute):
+    attr_type = ...  # type: Any
+    null = ...  # type: bool
+    def element_serialize(self, value): ...
+    def element_deserialize(self, value): ...
+    def serialize(self, value): ...
+    def deserialize(self, value): ...
+    def __init__(self, hash_key: bool = ..., range_key: bool = ..., null: Optional[bool] = ..., default: Optional[Union[Set[Text], Callable[..., Set[Text]]]] = ..., attr_name: Optional[Text] = ...) -> None: ...
+    def __set__(self, instance, value: Set[Text]): ...
+    def __get__(self, instance, owner) -> Set[Text]: ...
+
+class UnicodeAttribute(Attribute):
+    attr_type = ...  # type: Any
+    def __set__(self, instance, value: Text): ...
```

Comment:
Might make sense to make the base class Generic, so that you don't have to repeat set/get/init in every class.